ofctrl/ofTlvMap.go

```diff
@@ -134,24 +135,43 @@ type tlvMapMgr struct {
 
 func (m *tlvMapMgr) TLVMapReplyRcvd(ofSwitch *OFSwitch, status *TLVTableStatus) {
 	m.status = status
-	m.tlvCh <- struct{}{}
+	if m.tlvCh != nil {
+		close(m.tlvCh)
+	}
 }
 
 func (s *OFSwitch) requestTlvMap() error {
 	if s.tlvMgr == nil {
 		return nil
 	}
+	s.tlvMgr.tlvCh = make(chan struct{}, 1)
 	msg := openflow15.NewTLVTableRequest()
-	err := s.Send(msg)
+
+	var err error
+	defer func() {
+		if err != nil {
+			close(s.tlvMgr.tlvCh)
+		}
+		s.tlvMgr.tlvCh = nil
```

Comment:
I think this would lead to a race condition if `TLVMapReplyRcvd` happens to be executed when it timeouts or gets disconnected.

Reply:
Updated.